### PR TITLE
Image classification dataloader + deterministic output order

### PR DESCRIPTION
**Pull request by benartuso**
Also tweaked the CelebA files - image paths were wrong

### CODE REVIEW

File: oxen/python/oxen/ops/create_label_map.py [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+
+class CreateLabelMap(oxen.Op):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def call(self, args):
+        lines = args[0].split("\n")
+        canonical_labels = set(lines)
+        data_labels = set(args[1])
```
Comment (gschoeni) [inline oxen/python/oxen/ops/create_label_map.py:11]: We might want to add some comments or checks for what arg[0] and arg[1] are expected to be

File: oxen/python/oxen/ops/read_image_dir.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def call(self, args):
+        image_data = []
+        prefix = args[0]
+        for path in tqdm(args[1]):
+            img = cv2.imread(f"{prefix}/{path}")
+            image_data.append(img)
+        # TODO: Todo: handle differing image shapes, or leave that to user?
```
Comment (gschoeni) [inline oxen/python/oxen/ops/read_image_dir.py:17]: Yeah I could see this being in the constructor, good for this version

File: oxen/tests/test_image_classification_loader.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    )
+    data, labels, mapper = loader.run()   
+
+    assert data.shape == (5, 218, 178, 3), "Data not returned in expected shape"
+    assert labels.shape == (5,)
+    assert len(mapper.items()) == 3
+
+
+    # Test ease of use with pytorch
+    torch_data = TensorDataset(torch.from_numpy(data), torch.from_numpy(labels))
```
Comment (gschoeni) [inline oxen/tests/test_image_classification_loader.py:36]: We could also have class that is a torch.Dataset, that has __getitem__ and __len__ https://pytorch.org/tutorials/beginner/basics/data_tutorial.html

But this is good for now too

**Review by gschoeni — COMMENTED** [review]
Let's comment what the args you expect are, and then I think it's g2g

File: oxen/tests/test_image_classification_loader.py (hunk 3 above)
Comment (benartuso) [reply oxen/tests/test_image_classification_loader.py:36]: made a ticket 👍

File: oxen/python/oxen/ops/read_image_dir.py (hunk 2 above)
Comment (benartuso) [reply oxen/python/oxen/ops/read_image_dir.py:17]: ticked added 👍

File: oxen/python/oxen/ops/create_label_map.py (hunk 1 above)
Comment (benartuso) [reply oxen/python/oxen/ops/create_label_map.py:21]: updated!

**Comment by gschoeni** [comment]
LGTM 👍

**Pull request merged**